hawkular-command-gateway-war/src/main/java/org/hawkular/cmdgw/command/ws/server/AbstractGatewayWebSocket.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.cmdgw.command.ws.server;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import javax.inject.Inject;
+import javax.websocket.CloseReason;
+import javax.websocket.CloseReason.CloseCodes;
+import javax.websocket.OnMessage;
+import javax.websocket.Session;
+
+import org.hawkular.accounts.websocket.Authenticator;
+import org.hawkular.accounts.websocket.WebsocketAuthenticationException;
+import org.hawkular.bus.common.BasicMessage;
+import org.hawkular.bus.common.BasicMessageWithExtraData;
+import org.hawkular.cmdgw.GatewayLoggers;
+import org.hawkular.cmdgw.MsgLogger;
+import org.hawkular.cmdgw.api.ApiDeserializer;
+import org.hawkular.cmdgw.api.AuthMessage;
+import org.hawkular.cmdgw.api.Authentication;
+import org.hawkular.cmdgw.api.GenericErrorResponse;
+import org.hawkular.cmdgw.api.GenericErrorResponseBuilder;
+import org.hawkular.cmdgw.api.UiSessionOrigin;
+import org.hawkular.cmdgw.command.NoCommandForMessageException;
+import org.hawkular.cmdgw.command.ws.WsCommand;
+import org.hawkular.cmdgw.command.ws.WsCommandContext;
+import org.hawkular.cmdgw.command.ws.WsCommandContextFactory;
+import org.hawkular.cmdgw.command.ws.WsCommands;
+import org.hawkular.cmdgw.command.ws.WsEndpoints;
+
+/**
+ * A common parent for {@link FeedWebSocket} and {@link UIClientWebSocket}.
+ *
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ */
+public abstract class AbstractGatewayWebSocket {
+    private static final MsgLogger log = GatewayLoggers.getLogger(AbstractGatewayWebSocket.class);
+
+    @Inject
+    protected Authenticator authenticator;
+
+    @Inject
+    protected WsCommandContextFactory commandContextFactory;
+
+    protected final String endpoint;
+    @Inject
+    protected WsCommands wsCommands;
+    @Inject
+    protected WsEndpoints wsEndpoints;
+
+    public AbstractGatewayWebSocket(String endpoint) {
+        super();
+        this.endpoint = endpoint;
```

Comment:
this endpoint string looks to be either "ui/ws" or "feed/{feedId}" (literally with {feedId}). Not sure if this will be helpful.